crates/goose-mcp/src/developer/mod.rs

```diff
@@ -259,83 +259,168 @@ impl DeveloperRouter {
-        let (text_editor_desc, str_replace_command) = if let Some(ref editor) = editor_model {
-            (
-                formatdoc! {r#"
-                Perform text editing operations on files.
-
-                The `command` parameter specifies the operation to perform. Allowed options are:
-                - `view`: View the content of a file.
-                - `write`: Create or overwrite a file with the given content
-                - `edit_file`: Edit the file with the new content.
-                - `insert`: Insert text at a specific line location in the file.
-                - `undo_edit`: Undo the last edit made to a file.
-
-                To use the write command, you must specify `file_text` which will become the new content of the file. Be careful with
-                existing files! This is a full overwrite, so you must include everything - not just sections you are modifying.
-
-                To use the edit_file command, you must specify both `old_str` and `new_str` - {}.
-
-                To use the insert command, you must specify both `insert_line` (the line number after which to insert, 0 for beginning) 
-                and `new_str` (the text to insert).
-            "#, editor.get_str_replace_description()},
-                "edit_file",
-            )
-        } else {
-            (indoc! {r#"
-                Perform text editing operations on files.
-
-                The `command` parameter specifies the operation to perform. Allowed options are:
-                - `view`: View the content of a file.
-                - `write`: Create or overwrite a file with the given content
-                - `str_replace`: Replace a string in a file with a new string.
-                - `insert`: Insert text at a specific line location in the file.
-                - `undo_edit`: Undo the last edit made to a file.
-
-                To use the write command, you must specify `file_text` which will become the new content of the file. Be careful with
-                existing files! This is a full overwrite, so you must include everything - not just sections you are modifying.
-
-                To use the str_replace command, you must specify both `old_str` and `new_str` - the `old_str` needs to exactly match one
-                unique section of the original file, including any whitespace. Make sure to include enough context that the match is not
-                ambiguous. The entire original string will be replaced with `new_str`.
-
-                To use the insert command, you must specify both `insert_line` (the line number after which to insert, 0 for beginning) 
-                and `new_str` (the text to insert).
-            "#}.to_string(), "str_replace")
+        // Create text editor tool with different descriptions and schemas based on editor API configuration
+        let (text_editor_desc, schema_properties) = match &editor_model {
+            Some(EditorModel::MorphLLM(editor)) => {
```

Comment:
small one - but may refactor this so there is no dependence on specific editor models - can use the replace_description() type of thing to sub it in accordingly (and can probably similarly get the instructions tool added)